packages/playwright-core/src/server/injected/selectorGenerator.ts

```diff
@@ -205,12 +205,12 @@ function buildTextCandidates(injectedScript: InjectedScript, element: Element, i
     return [];
   const candidates: SelectorToken[] = [];
 
-  const escaped = escapeForTextSelector(text, false, true);
+  const escaped = escapeForTextSelector(text, false);
 
   if (isTargetNode)
-    candidates.push({ engine: 'text', selector: escaped, score: 10 });
+    candidates.push({ engine: 'internal:text', selector: escaped, score: 10 });
 
-  if (escaped === text) {
+  if (!escaped.startsWith('/')) {
```

Comment:
This check does not guarantee that `has-text("{text}")` is a valid CSS selector now. Perhaps we should `cssEscape(text)` in line 217? Or replace it all with `internal:has="internal:text=..."` like we do with `hasText` option?